WordPress/Classes/ViewRelated/Views/BottomSheet/BottomSheetPresentationController.swift

```diff
@@ -0,0 +1,251 @@
+
+import UIKit
+
+// MARK: - BottomSheetPresentationControllerConstants
+
+private struct BottomSheetPresentationControllerConstants {
+    static let cornerRadius         = CGFloat(16)
+    static let transitionDuration   = TimeInterval(0.3)
+}
+
+// MARK: - BottomSheetPresentationController
+
+///
+/// BottomSheetPresentationController manages presentation of a view controller like a bottom sheet.
+///
+/// Adapted from AAPLCustomPresentationController, Custom View Controller Presentations and Transitions
+/// https://developer.apple.com/library/archive/samplecode/CustomTransitions/Introduction/Intro.html#//apple_ref/doc/uid/TP40015158
+///
+class BottomSheetPresentationController: UIPresentationController {
+
+    // MARK: Properties
+
+    private(set) var dimmingView: UIView?
+    private(set) var presentationWrappingView: UIView?
+
+    // MARK: UIPresentationController
+
+    override init(presentedViewController: UIViewController, presenting presentingViewController: UIViewController?) {
+
+        super.init(presentedViewController: presentedViewController, presenting: presentingViewController)
+        presentedViewController.modalPresentationStyle = .custom
+    }
+
+    override var presentedView: UIView? {
+        return self.presentationWrappingView
+    }
+
+    override func presentationTransitionWillBegin() {
```

Comment:
When I read this method, I see several blocks clearly delimited, and separated by a white line. That, to me, usually indicates that the method is doing more than one thing, which is fine, but usually each and every one of those things the method is doing can be encapsulated into a separate method.

I guess, my question would be: would it make sense to encapsulate each block of logic in a separate method? A certain author I won't name called that "single level of abstraction", if I recall correctly. The way I interpret it, when reading `presentationTransitionWillBegin` I'd find more useful to find statements of the same level of abstraction (i.e. `preparePresentationWrapper()` `preparePresentedViewController()` `prepareDimmingView()`). If I need to dig deep into any of those, their implementation would be a command+click away